src/com/trilead/ssh2/crypto/dh/DhGroupExchange.java

```diff
@@ -88,9 +87,33 @@ public void setF(BigInteger f)
 	}
 
 	public byte[] calculateH(byte[] clientversion, byte[] serverversion, byte[] clientKexPayload,
-			byte[] serverKexPayload, byte[] hostKey, DHGexParameters para)
+ 			byte[] serverKexPayload, byte[] hostKey, DHGexParameters para)
+	{
+		HashForSSH2TypesNew hash = new HashForSSH2TypesNew("SHA1");
```

Comment:
Could you not just have done `return calculateH("SHA1", clientversion, serverversion, clientKexPayload, serverKexPayload, hostKey, para)` here, rather than duplicate the method body (I may not have explained my original directions clearly enough)?